AB#2685 Add documentation for automatic IAM creation (#711)
edgelesssys/constellation: docs/docs/workflows/config.md

> # Configure your cluster

Author: You definitely have a point there. The only thing that seems counter-intuitive to me with that approach is the path for a user setting up his cluster. He would go to create -> config -> create. Should we maybe add the config-generation step to the *config* workflow page and add a box to the create-step which indicates that before creating a cluster, a filled config file is needed (while pointing to the *config*-page from there)? Imo, grouping these 2 would make more sense.

> # Configure your cluster

Reviewer: sounds good!

> Before your cluster can be created, you need to configure the identity and access management (IAM) for your CSP and choose machine types for your cluster.

Reviewer: ```suggestion
Before you can create your cluster, you need to configure the identity and access management (IAM) for your cloud service provider (CSP) and choose machine types for the nodes.
```
prefer active voice
spell out acronyms on first use on a page (except for those that are more well known than the spelled-out term)

## Choosing VM types

Constellation supports the following VM types:

> <tabs groupID="csp">

Reviewer: ```suggestion
<tabs groupId="csp">
```
Docusaurus seems to be case sensitive with this. At least this change fixes the tabs not being in sync.

</tabItem>
</tabs>

Fill the desired VM type into the **instanceType** field in the `constellation-conf.yml` file.

## Creating an IAM configuration

IAM configuration for your cluster can be created automatically using the `constellation iam create` command.
<tabs groupId="csp">
<tabItem value="azure" label="Azure">

You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session.

```bash
constellation iam create azure --region=westus --resourceGroup=constellTest --servicePrincipal=spTest
```

:::info
This command creates IAM configuration on the Azure region `westus` creating a new resource group `constellTest` and a new service principal `spTest`.
:::

Note that CVMs are currently only supported in a few regions, check [Azure's products available by region](https://azure.microsoft.com/en-us/global-infrastructure/services/?products=virtual-machines&regions=all). These are:
  * `westus`

>   * `eastus`

Reviewer: IMO the regions are essential enough to also mention them in the *first steps*. We should prevent users from making a mistake there.

```bash
constellation iam create gcp --projectID=yourproject-12345 --zone=europe-west1-a --serviceAccountID=constell-test
```

:::info
This command creates IAM configuration in the GCP project `yourproject-12345` on the GCP zone `europe-west1-a` creating a new service account `constell-test`.
:::

Note that only regions offering CVMs of the `N2D` series are supported. You can find a [list of all regions in Google's documentation](https://cloud.google.com/compute/docs/regions-zones#available), which you can filter by machine type `N2D`.

Paste the output into the corresponding fields of the `constellation-conf.yaml` file.

</tabItem>
<tabItem value="aws" label="AWS">

You must be authenticated with the [AWS CLI](https://aws.amazon.com/en/cli/) in the shell session.

```bash
constellation iam create aws --zone=eu-central-1a --prefix=constellTest
```

:::info
This command creates IAM configuration for the AWS zone `eu-central-1a` using the prefix `constellTest` for all named resources being created.
:::

Paste the output into the corresponding fields of the `constellation-conf.yaml` file.

Constellation OS images are currently replicated to the following regions:

>     * `eu-central-1`
>     * `us-east-2`
>     * `ap-south-1`

Reviewer: ```suggestion
* `eu-central-1`
* `us-east-2`
* `ap-south-1`
```
otherwise it's not rendered correctly

If you require the OS image to be available in another region, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md&title=Support+new+AWS+image+region:+xx-xxxx-x).

You can find a list of all [regions in AWS's documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions).

</tabItem>
</tabs>

<details>

> <summary>Alternatively, you can manually create the IAM configuration on your CSP.</summary>

Reviewer: ```suggestion
<summary>Alternatively, you can manually create the IAM configuration on your CSP.</summary>
The following describes the configuration fields and how you obtain the required information or create the required resources.
```
This introductory sentence shall make clear why the following is a list of config field names.

<tabs groupId="csp">

> <tabItem value="azure" label="Azure (manually)">

Reviewer: ```suggestion
<tabItem value="azure" label="Azure">
```
as this is now separated, we can omit it

  Alternatively, you can create the AWS profile with a tool of your choice. Use the JSON policy in [main.tf](https://github.com/edgelesssys/constellation/tree/release/v2.2/hack/terraform/aws/iam/main.tf) in the resource `aws_iam_policy.control_plane_policy`.

* **iamProfileWorkerNodes**: The name of an IAM instance profile attached to all worker nodes.

  Use the [provided Terraform script](https://github.com/edgelesssys/constellation/tree/release/v2.2/hack/terraform/aws/iam) to generate the necessary profile. The profile name will be provided as Terraform output value: `worker_nodes_instance_profile`.

  Alternatively, you can create the AWS profile with a tool of your choice. Use the JSON policy in [main.tf](https://github.com/edgelesssys/constellation/tree/release/v2.2/hack/terraform/aws/iam/main.tf) in the resource `aws_iam_policy.worker_node_policy`.

</tabItem>

</tabs>
</details>

Now that you've configured your CSP, you can [create your cluster](./create.md).

## Deleting an IAM configuration

Created IAM configurations can be kept and reused for new clusters. Alternatively, you can also delete them if they aren't being used anymore.

> ### Prerequisites

Reviewer: ```suggestion
**Prerequisites:**
```
otherwise it creates an entry in the sidebar on the right and then we would need another heding for the actual commands

> * [HashiCorp's Terraform](https://developer.hashicorp.com/terraform/downloads?product_intent=terraform) is installed on your machine.

Reviewer: ```suggestion
* [Terraform](https://developer.hashicorp.com/terraform/downloads) is installed on your machine.
```
nit/optional: IMO the product name is descriptive enough

* Access to the `terraform.tfstate` file created by the `constellation iam create` command.

You can delete the IAM configuration using the following commands:
```bash
# Navigate to the directory containing the terraform.tfstate file
cd constellation-iam-terraform
# Destroy the IAM configuration via Terraform
terraform destroy

> # Confirm deletion via typing "yes"

Reviewer: ```suggestion
# Confirm deletion by typing "yes"
```

</tabItem>
</tabs>

Fill the desired VM type into the **instanceType** field in the `constellation-conf.yml` file.

## Creating an IAM configuration

You can create an IAM configuration for your cluster automatically using the `constellation iam create` command.
<tabs groupId="csp">
<tabItem value="azure" label="Azure">

You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session.

```bash
constellation iam create azure --region=westus --resourceGroup=constellTest --servicePrincipal=spTest
```

:::info
This command creates IAM configuration on the Azure region `westus` creating a new resource group `constellTest` and a new service principal `spTest`.

> :::

Reviewer: we should "unbox" the three infos on this page, too
